src/coinbase/smart_contract.ts

```diff
@@ -1,10 +1,220 @@
+  SmartContractOptions,
+  SmartContractType,
+  NFTContractOptions,
+  TokenContractOptions,
+  TransactionStatus,
+} from "./types";
 import { Coinbase } from "./coinbase";
+import { delay } from "./utils";
+import { TimeoutError } from "./errors";
 import { ContractEvent } from "./contract_event";
 
 /**
  * A representation of a SmartContract on the blockchain.
  */
 export class SmartContract {
+  private model: SmartContractModel;
+
+  /**
+   * Creates a new SmartContract instance.
+   *
+   * @param contractModel - The SmartContract model from the API.
+   */
+  constructor(contractModel: SmartContractModel) {
+    if (!contractModel) {
+      throw new Error("SmartContract model cannot be empty");
+    }
+    this.model = contractModel;
+  }
+
+  /**
+   * Converts a SmartContractModel into a SmartContract object.
+   *
+   * @param contractModel - The SmartContract model object.
+   * @returns The SmartContract object.
+   */
+  public static fromModel(contractModel: SmartContractModel): SmartContract {
+    return new SmartContract(contractModel);
+  }
+
+  /**
+   * Returns the ID of the SmartContract.
+   *
+   * @returns The SmartContract ID.
+   */
+  public getId(): string {
+    return this.model.smart_contract_id;
+  }
+
+  /**
+   * Returns the Network ID of the SmartContract.
+   *
+   * @returns The Network ID.
+   */
+  public getNetworkId(): string {
+    return this.model.network_id;
+  }
+
+  /**
+   * Returns the Wallet ID that deployed the smart contract.
+   *
+   * @returns The Wallet ID.
+   */
+  public getWalletId(): string {
+    return this.model.wallet_id;
+  }
+
+  /**
+   * Returns the Contract Address of the smart contract.
+   *
+   * @returns The Contract Address.
+   */
+  public getContractAddress(): string {
+    return this.model.contract_address;
+  }
+
+  /**
+   * Returns the Deployer Address of the smart contract.
+   *
+   * @returns The Deployer Address.
+   */
+  public getDeployerAddress(): string {
+    return this.model.deployer_address;
+  }
+
+  /**
+   * Returns the Type of the smart contract.
+   *
+   * @returns The Smart Contract Type.
+   */
+  public getType(): SmartContractType {
+    switch (this.model.type) {
+      case SmartContractTypeModel.Erc20:
+        return SmartContractType.ERC20;
+      case SmartContractTypeModel.Erc721:
+        return SmartContractType.ERC721;
+      default:
+        throw new Error(`Unknown smart contract type: ${this.model.type}`);
+    }
+  }
+
+  /**
+   * Returns the Options of the smart contract.
+   *
+   * @returns The Smart Contract Options.
+   */
+  public getOptions(): SmartContractOptions {
+    if (this.getType() === SmartContractType.ERC20) {
+      return this.model.options as TokenContractOptions;
+    }
+    return this.model.options as NFTContractOptions;
+  }
+
+  /**
+   * Returns the ABI of the smart contract.
+   *
+   * @returns The ABI as a JSON-encoded string.
+   */
+  public getAbi(): string {
+    return this.model.abi;
```

Comment:
Do we want to `JSON.parse` this object and return the object?

https://github.com/coinbase/coinbase-sdk-nodejs/blob/44abe9858c58664c799dade7bbf75bb23f14a1a6/src/coinbase/contract_invocation.ts#L110-L114

Reply:
good catch, fixed in updated PR!